test/e2e/kustomize/api-server.yaml

```diff
@@ -12,12 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-- op: replace
-  path: /spec/ports/0
-  value:
-    name: server
-    port: 8080
-    nodePort: 30000
 - op: add
-  path: /spec/type
-  value: NodePort
+  path: /spec/template/spec/containers/0/env/-
+  value:
+    name: TLS_HOSTNAME_OVERRIDE
+    value: tekton-results-api-service.tekton-pipelines.svc.cluster.local
```

Comment:
This required because the the gRPC gateway code Dials a connection to existing gRPC server locally. Hence it gets the domain name `localhost`, but the endpoint is using a certificate with DNS record for the kubernetes service, so it doesn't allow the connection. Passing the override allows to set the `authority` header while dialling the gRPC server locally from the gateway code.
https://github.com/tektoncd/results/blob/965033be9aaf79c943123ceeabe858e11787de07/proto/v1alpha2/results_go_proto/api.pb.gw.go#L1176